typescript-for-haskellers/article.md

```diff
@@ -0,0 +1,916 @@
+# TypeScript for Haskellers
+
+Web app nowadays is mandatory part of every modern application.
+It doesn't matter how small or complex it is.
+From one click apps with pictures converting to photoshop everyone wants fast and easy access to it and Web is where you can give it to your users or even for you with the easiest way.
+
+In Serokell we use [TypeScript](https://www.typescriptlang.org/) for writing web pages.
+You can read [Why You Should Choose TypeScript Over JavaScript](https://serokell.io/blog/why-typescript) in our blog.
+And as our main language is Haskell, in this article we want to explain how knowledge from it will help you to write TypeScript code.
+
+## Why TypeScript?
+
+Historically, any new language which allow you to write web applications translate its code to JavaScript.
+And there are a lot of such languages and language extensions: TypeScript, Kotlin JS, CoffeeScript, Scala JS, Babel, ClojureScript, etc.
+Generally you can write web applications almost on every language.
+
+Because we like Haskell, maybe we can write our Web application on functional languages or even on Haskell? And the answer is yes.
+There is strongly-typed PureScript with Haskell like syntax, purely functional Elm, Reflex framework for Haskell with GHCJS, F# with Fabel.
+
+Unfortunately they don't give us proper flexibility in working with JS libraries which number is enormous and take more time on fixing related problems.
+Additionally, customers will be locked in this solution when they will need to maintenance such applications.
+On the other hand TypeScript with a huge community and modern technologies from the box is a great decision.
+
+In this article we don't have a goal to cover all similarities or differences.
+We also will not say about disadvantages of TypeScript type system against Haskell or how it works under the hood.
+Our goal is to show what and how you can use types in TypeScript.
+
+## Base concepts
+
+Unlike Haskell in TypeScript types are not the center part of the language.
+They just help programmers to write stricter and more understandable code, making their life easier.
+You can use types like `any` and nobody pushes you into strict frames.
```

Comment:
Do you mean anything specific by "strict frames"? I don't get it.

Reply:
Looks like I need word `framework` here